tutorials/shaders/shader_reference/shading_language.rst

```diff
@@ -839,6 +839,191 @@ The syntax also supports subgroups (it's not mandatory to declare the base group
 
     group_uniforms MyGroup.MySubgroup;
 
+Global uniforms
+^^^^^^^^^^^^^^^
+
+Sometimes, you want to modify a parameter in a lot of instances of a material.
+This takes a lot of work as all these instances need to be tracked and the
+uniform needs to be set for each of them.
+
+Example use cases include:
+
+1. **Providing player information to all relevant shaders.** You may want to let
+   your shaders know where the player is. As an example, the grass and some
+   bushes will bend when the player is close, simulating being pushed.
+2. **Modifying the world without actually modifying it.** Modern games have a lot
+   of grass, yet you can do precise cuts to it, burn it, etc. How do they do it?
+   Is every bit of grass a mesh and an object? Not really, games generally just
+   use a texture aligned to the world (that often moves with you), with different
+   colors meaning different things. In this case they can use texture channels to
+   specify that the grass has been cut, burnt, etc. The vertex shader can make
+   the vertices disappear, or change the color based on this global uniform
+   texture.
+3. **Controlling the weather and environment.** Nowadays, games have realistic
+   weather, with leaves and trees moving with the wind. Do they make animations
+   for each piece of grass? Not really, you can save the wind strength and
+   direction in a global uniform, then procedurally move the foliage with
+   different speed and intensity. This can even affect particle system shaders,
+   making fire and smoke propagate in the direction of the wind. Shooters often
+   have what is called a turbulence texture, which is a texture that contains
+   the local wind direction around the player. All relevant shaders can access
+   it and it's what allows explosions to move trees and grass around.
+4. **Making regular materials interact with the environment.** This is often
+   overlooked. We think materials are edited and that's it, but that's not
+   always the case. Imagine it starts raining and you want your objects to
+   become wet. You need to change the metallic and roughness factors. For
+   floors, you want to draw droplets splashing. For walls, you want to draw the
+   water pouring down. You could control this with regular uniforms, but it
+   would be more time-consuming to set this on every material when it starts
+   raining.
+5. **Providing team colors.** In team-based games, you can use global uniforms
+   to provide "friendly" and "enemy" colors that can automatically be used
+   everywhere relevant (player materials, map textures, special effects, GUI
+   elements, ...). This can be used to let players choose team colors freely, for
+   better accessibility. The "friendly" and "enemy" states themselves can be
+   determined by using a :ref:`per-instance uniform
+   <doc_shading_language_per_instance_uniforms>` that indicates the instance's
+   current team.
```

Comment:
I would massively simplify this whole block. We don't need to provide a detailed list to the user about where they might find this helpful. Such a list makes sense for a blog post where we are promoting a new feature, but this is an API reference.

```suggestion
Global uniforms are especially useful for environmental effects that affect many objects in a scene like having foliage bend when the player is near or having objects move with the wind. 
```